holidays/countries/thailand.py

```diff
@@ -12,156 +12,880 @@
+            (JUN, 13, rama_ix_sixty_accession_en),
+            (SEP, 20, thai_military_emergency_lockdown_en),
+        ),
+        2007: ((DEC, 24, thai_election_in_lieu_en),),
+        2009: (
+            (JAN, 2, thai_bridge_public_holiday_en),
+            (APR, 10, thai_political_emergency_lockdown_en),
+            (APR, 16, thai_political_emergency_lockdown_en),
+            (APR, 17, thai_political_emergency_lockdown_en),
+            (JUL, 6, thai_bridge_public_holiday_en),
+        ),
+        2010: (
+            (MAY, 20, thai_bridge_public_holiday_en),
+            (MAY, 21, thai_bridge_public_holiday_en),
+            (AUG, 13, thai_bridge_public_holiday_en),
+        ),
+        2011: (
+            (MAY, 16, thai_bridge_public_holiday_en),
+            (OCT, 27, thai_flood_2011_emergency_lockdown_en),
+            (OCT, 28, thai_flood_2011_emergency_lockdown_en),
+            (OCT, 31, thai_flood_2011_emergency_lockdown_en),
+        ),
+        2012: ((APR, 9, thai_bridge_public_holiday_en),),
+        2013: ((DEC, 30, thai_bridge_public_holiday_en),),
+        2014: ((AUG, 11, thai_bridge_public_holiday_en),),
+        2015: (
+            (JAN, 2, thai_bridge_public_holiday_en),
+            (MAY, 4, thai_bridge_public_holiday_en),
+        ),
+        2016: (
+            (MAY, 6, thai_bridge_public_holiday_en),
+            (JUL, 18, thai_bridge_public_holiday_en),
+            (OCT, 14, rama_ix_mourning_en),
+        ),
+        2017: ((OCT, 26, rama_ix_cremation_en),),
+        2019: ((MAY, 6, rama_x_coronation_celebrations_en),),
+        2020: (
+            (JUL, 27, songkran_festival_in_lieu_covid_en),
+            (SEP, 4, songkran_festival_in_lieu_covid_en),
+            (SEP, 7, songkran_festival_in_lieu_covid_en),
+            (NOV, 19, thai_bridge_public_holiday_en),
+            (NOV, 20, thai_bridge_public_holiday_en),
+            (DEC, 11, thai_bridge_public_holiday_en),
+        ),
+        2021: (
+            (FEB, 12, thai_bridge_public_holiday_en),
+            (APR, 12, thai_bridge_public_holiday_en),
+            (SEP, 24, thai_bridge_public_holiday_en),
+        ),
+        2022: (
+            (JUL, 15, thai_bridge_public_holiday_en),
+            (JUL, 29, thai_bridge_public_holiday_en),
+            (OCT, 14, thai_bridge_public_holiday_en),
+            (DEC, 30, thai_bridge_public_holiday_en),
+        ),
+        2023: ((MAY, 5, thai_bridge_public_holiday_en),),
+    }
+
+    def __init__(self, **kwargs):
+        HolidayBase.__init__(self, **kwargs)
 
-        # The Queen Sirikit's Birthday.
-        add_holiday(date(year, AUG, 12), "The Queen Mother's Birthday")
+    def _populate(self, year):
+        # DEFAULT
+        def add_holiday(dt, holiday_name) -> None:
+            # Only add if current year
+            # This is here as a stub for future islamic consecutive holidays
+            # Which can stradle across gregorian years in southern region
```

Comment:
Ah, it's for future, I understand now. You have islamic holidays also... I like this country more and more :))